### PR TITLE
Added example and documentation of fakes.

### PR DESCRIPTION
**Pull request by mwhittaker**
In Issue #419, @Markbnak pointed out that our documentation on fakes is a little lacking. I agree. This PR adds a simple example of faking and adds documentation to the website on faking.

### CODE REVIEW

File: examples/fakes/clock_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"context"
+	"testing"
+
+	"github.com/ServiceWeaver/weaver/weavertest"
+)
+
+// fakeclock is a fake implementation of the Clock component that always
+// returns a fixed time.
+type fakeclock struct {
```
Comment (spetrovic77) [inline examples/fakes/clock_test.go:26]: nit: `fakeClock`?

File: website/docs.md [hunk 2]
```diff
@@ -1396,6 +1398,57 @@ func TestAdd(t *testing.T) {
...
 
+## Fakes
+
+You can replace a component implementation with a fake implementation in a test
+using [`weavertest.Fake`][weavertest.Fake]. Here's an example where we replace
+the real implementation of a `Clock` component with a fake implementation that
+always returns a fixed time.
+
+```go
+// fake is a fake implementation of the Clock component.
```
Comment (spetrovic77) [inline website/docs.md:1409]: fakeClock

File: website/docs.md [hunk 3]
```diff
@@ -1396,6 +1398,57 @@ func TestAdd(t *testing.T) {
...
+type fakeclock struct {
+    now int64
+}
+
+// Now implements the Clock component interface. It returns the current time, in
+// microseconds since the unix epoch.
```
Comment (spetrovic77) [inline website/docs.md:1415]: comma after `microseconds`.

**Review by spetrovic77 — APPROVED** [review]
Looks great, thanks for adding this section!

File: examples/fakes/clock_test.go (hunk 1 above)
Comment (mwhittaker) [reply examples/fakes/clock_test.go]: Done.

File: website/docs.md (hunk 2 above)
Comment (mwhittaker) [reply website/docs.md]: Done.

File: website/docs.md (hunk 3 above)
Comment (mwhittaker) [reply website/docs.md]: Done.

**Review by mwhittaker — COMMENTED** [review]
Thanks Srdjan!

**Pull request merged**